src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos, getUser } from './api';
+import { User } from './types/User';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loadingTodo, setLoadingTodo] = useState(true);
+  const [loadingUser, setLoadingUser] = useState(true);
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const [selectedOption, setSelectedOption] = useState<Filter>(Filter.ALL);
+  const [query, setQuery] = useState<string>('');
+
+  function handleSelectedOption(event: React.ChangeEvent<HTMLSelectElement>) {
+    setSelectedOption(event.target.value as Filter);
```

Comment:
The `handleSelectedOption` function casts the `event.target.value` to `Filter` using `as Filter`. This might cause runtime errors if the value is not one of the `Filter` enum values. Consider validating the value before casting or using a safer method to handle this conversion.